src/api/apollo.js

```diff
@@ -1,48 +1,150 @@
+	},
+	LoanChannel: {
+		keyFields: ['id'],
+	},
+	RecLoanChannel: {
+		keyFields: ['id'],
+	},
+	LoanPartner: {
+		fundraisingPercent: {
+			read(_, { readField }) {
+				return LoanPartner.fundraisingPercent(readField('loan'));
+			}
+		},
+		fundraisingTimeLeft: {
+			read(_, { readField }) {
+				return LoanPartner.fundraisingTimeLeft(readField('loan'));
+			}
+		},
+		unreservedAmount: {
+			read(_, { readField }) {
+				return LoanPartner.unreservedAmount(readField('loan'));
+			}
+		},
+		fundraisingTimeLeftMilliseconds: {
+			read(_, { readField }) {
+				return LoanPartner.fundraisingTimeLeftMilliseconds(readField('loan'));
+			}
+		},
+	},
+	LoanDirect: {
+		fundraisingPercent: {
+			read(_, { readField }) {
+				return LoanDirect.fundraisingPercent(readField('loan'));
+			}
+		},
+		fundraisingTimeLeft: {
+			read(_, { readField }) {
+				return LoanDirect.fundraisingTimeLeft(readField('loan'));
+			}
+		},
+		unreservedAmount: {
+			read(_, { readField }) {
+				return LoanDirect.unreservedAmount(readField('loan'));
+			}
+		},
+		fundraisingTimeLeftMilliseconds: {
+			read(_, { readField }) {
+				return LoanDirect.fundraisingTimeLeftMilliseconds(readField('loan'));
+			}
+		},
+	},
+	loanSearchState: {
+		keyFields: ['id'],
+	},
+	Manifest: {
+		keyFields: ['id'],
+	},
+	shop: {
+		keyFields: ['id'],
+	},
+	Setting: {
+		keyFields: ['key'],
+	},
+	Query: {
+		fields: {
+			community: {
+				merge: true
+			},
+			general: {
+				merge: true
+			},
+			// The ml field doesn't have an ID
+			ml: {
+				merge: true,
+			},
+			// The contentful field doesn't have an ID
+			contentful: {
+				merge: true,
+			},
+			// The lend field doesn't have an ID
+			lend: {
+				merge: true,
+			},
+			// The my field doesn't have an ID
+			my: {
+				merge: true,
+			},
+			getCategories: {
+				merge: true
+			},
+			mySubscriptions: {
+				merge: true
+			},
+			fundraisingLoans: {
+				merge: true
+			}
+		},
+	},
+	Mutation: {
+		AutolendingMutation: {
+			merge: true
+		}
+	}
+};
 
 export default function createApolloClient({
-	appConfig,
+	// appConfig,
 	cookieStore,
 	kvAuth0,
-	types = [],
+	types,
 	uri,
 	fetch
 }) {
-	const cache = new InMemoryCache({
-		fragmentMatcher: new IntrospectionFragmentMatcher({
-			introspectionQueryResultData: {
-				__schema: { types }
-			}
-		}),
-		// Return a custom cache id for types that don't have an id field
-		dataIdFromObject: object => {
-			if (object.__typename === 'Setting' && object.key) return `Setting:${object.key}`;
-			return defaultDataIdFromObject(object);
-		},
-		// Use a simpler underlying cache for server renders
-		resultCaching: typeof window !== 'undefined',
-		// Block modifying cache results outside of normal operations
-		// see https://github.com/apollographql/apollo-client/pull/4543
-		freezeResults: true,
+	// initialize local state resolvers
+	// TODO: convert resolvers into typePolicies
+	// const { resolvers } = initState({ appConfig, cookieStore, kvAuth0 });
```

Comment:
Is more work needed here for resolvers or can this commented out code be removed (and the commented out `appConfig` above?